asab/config.py

```diff
@@ -283,7 +283,12 @@ class _Interpolation(configparser.ExtendedInterpolation):
 	def before_read(self, parser, section, option, value):
 		# Expand environment variables
 		if '$' in value:
+
 			os.environ['THIS_DIR'] = os.path.abspath(parser._load_dir_stack[-1])
+
+			# Hostname cannot be interpolated here, because it is not known
+			os.environ['HOSTNAME'] = "{HOSTNAME}"
```

Comment:
This doesn't make much sense honestly.

Reply:
@ateska Unfortunately not, if I remove it from here, Python's config parser will complain:

```
Traceback (most recent call last):
  File "logging-file.py", line 24, in <module>
    app = MyApplication()
  File "/home/pcerny/Workspace/base/asab/asab/abc/singleton.py", line 14, in __call__
    cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
  File "/home/pcerny/Workspace/base/asab/asab/application.py", line 84, in __init__
    self.Logging = Logging(self)
  File "/home/pcerny/Workspace/base/asab/asab/log.py", line 48, in __init__
    file_path = Config["logging:file"]["path"]
  File "/usr/lib/python3.8/configparser.py", line 1255, in __getitem__
    return self._parser.get(self._name, key)
  File "/usr/lib/python3.8/configparser.py", line 799, in get
    return self._interpolation.before_get(self, section, option, value,
  File "/usr/lib/python3.8/configparser.py", line 456, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/lib/python3.8/configparser.py", line 507, in _interpolate_some
    raise InterpolationMissingOptionError(
configparser.InterpolationMissingOptionError: Bad value substitution: option 'path' in section 'logging:file' contains an interpolation key 'HOSTNAME' which is not a valid option name. Raw value: './${HOSTNAME}/log.log'
```

Reply:
Then it has to be part of the "setter" to `app.Hostname`

it == `os.environ['HOSTNAME'] = new_hostname`

Reply:
True that, I fixed it. Thanks.